tests/ComTests.cpp

```diff
@@ -3058,4 +3060,175 @@ TEST_CASE("COMEnumerator", "[com][enumerator]")
 #pragma warning(pop)
 #endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && WIL_HAS_CXX_17
 
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
+#if (NTDDI_VERSION >= NTDDI_WINBLUE)
+#if defined(__cpp_impl_coroutine) || defined(__cpp_coroutines) || defined(_RESUMABLE_FUNCTIONS_SUPPORTED)
+
+#include <wil/cppwinrt_register_com_server.h>
+#include <winrt/windows.foundation.h>
+#include <windows.foundation.h>
+
+const winrt::guid CLSID_RPCTimeoutTestServer{L"CED2F47C-200B-476F-BFDC-D0D79B052AC6"};
+HANDLE g_hangHandle{nullptr};
+HANDLE g_doneHangingHandle{nullptr};
+
+TEST_CASE("com_timeout", "[com][com_timeout]")
+{
+    auto init = wil::CoInitializeEx_failfast();
+
+    // These test cases require a COM server that will use RPC for communication so that we can exercise
+    // RPC cancellation.  Additionally, this server needs to support the ability to control if it hangs
+    // or returns quickly.  The following class provides that functionality, using some global events to
+    // provide deterministic ordering of operations.
+    //
+    // Normally a server in the same process would not use RPC.  By marking this as non_agile and running
+    // it on an STA thread we are able to get RPC involved enough to test cancellation.

```

Comment:
Per previous comment, consider:
"To ensure the call goes through a proxy, the object must be non-agile and the call must cross between apartments (MTA -> STA in this case)".

Reply:
Copied the suggestion verbatim.